cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -811,6 +811,19 @@ impl TryFrom<PackageInfoCommon> for SearchResult {
     }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum ParsedSearch {
+    Clean(String),
+    VersionStripped(String),
+}
+
+pub fn parse_search_term(search_term: &str) -> ParsedSearch {
+    match search_term.split_once('@') {
```

Comment:
the #1562 parsing uses `rsplit_once` to allow for `@`s in the attribute path (it's still not without flaws bc now you _need_ to provide a version part as well)

Reply:
I cribbed this from (which I assume will also go away with pkgdb): https://github.com/flox/flox/blob/3789c408182f1b1e60b44f55f496f5939e9a41be/cli/flox-rust-sdk/src/models/search.rs#L192

Is it valid to have an `@` in the attribute path? What about in a version? Do you know of any examples in the wild of either?

Reply:
`nixpkgs#nodePackages.@angular/cli`

lots of nodePackages (that are automatically ingested)

Reply:
FWIW - In our test DB (7 live scrapes) I don't see any versions with `@`.

Reply:
Every instance I see of an `@` in the attr path, it starts with it. Also, since we index on the last tuple of the attr_path, searches for `nodePackages.@angular` would fail anyway.  Given `@angular` and we split will give an error as there is no search term, and `angular@<ver>` will work as expected.  So I don't think there is much effect on the behavior in those odd cases.  I think that's something we could follow up on later IMO.

Reply:
Thanks both. I think it makes sense to stick with `strip_once` then; I've pushed an additional unit test to document the behaviour and rationale.